client/src/cordova/android/OutlineAndroidLib/outline/src/main/java/org/outline/vpn/VpnTunnelService.java

```diff
@@ -307,14 +383,18 @@ public void onAvailable(Network network) {
       broadcastVpnConnectivityChange(TunnelStatus.CONNECTED);
       updateNotification(TunnelStatus.CONNECTED);
 
-      boolean isUdpSupported = vpnTunnel.updateUDPSupport();
+      boolean isUdpSupported = false;
+      if (VpnTunnelService.this.remoteDevice != null) {
+        isUdpSupported = VpnTunnelService.this.remoteDevice.updateUDPSupport();
+      }
       LOG.info(
           String.format("UDP support: %s -> %s", tunnelStore.isUdpSupported(), isUdpSupported));
+      // Why do we need to persist this? TODO(fortuna): remove.
       tunnelStore.setIsUdpSupported(isUdpSupported);
```

Comment:
When the OS just got started, Internet might not be ready (e.g., connecting Wifi/Cellular), so there's no way to actually test it?

Reply:
It's being tested above (`VpnTunnelService.this.remoteDevice.updateUDPSupport()`). Not sure I follow.

Reply:
Ah, I'm not commenting the code, I'm answering the question of your comment.